relayer/network_utils.go

```diff
@@ -86,30 +95,47 @@ func checkSufficientConnectedStake(
 			)
 			return err
 		}
-		if utils.CheckStakeWeightExceedsThreshold(
+
+		if !utils.CheckStakeWeightExceedsThreshold(
 			big.NewInt(0).SetUint64(connectedValidators.ConnectedWeight),
 			connectedValidators.ValidatorSet.TotalWeight,
 			maxQuorumNumerator,
 		) {
-			logger.Info(
-				"Connected to sufficient stake",
+			// Log details of each connected validator (nodeID and weight).
+			// This is useful for troubleshooting startup issues when the relayer fails to connect to sufficient stake.
+			if logger.Enabled(logging.Debug) {
```

Comment:
I would downgrade this one to be `verbose` level instead of `debug` since it's at runtime and on a per-request basis. But i'm also fine leaving it as is since we have had even louder debugs in the past.